crates/db_schema/src/impls/local_user_language.rs

```diff
@@ -1,22 +1,40 @@
-use crate::{newtypes::LocalUserId, source::local_user_language::*};
+use crate::{
+  newtypes::{LanguageId, LocalUserId},
+  source::{language::Language, local_user_language::*},
+};
 use diesel::{result::Error, PgConnection, RunQueryDsl, *};
 
 impl LocalUserLanguage {
+  /// Update the user's languages.
+  ///
+  /// If no language_id vector is given, it will show all languages
   pub fn update_user_languages(
```

Comment:
Refactored this to use an Optional vector of `LanguageId`. If non is given, it inserts every language for the user, so a left join and filter to the local_user_language table will return all possible posts.

Reply:
Seems a bit inefficient to store it this way, but if its a problem we can optimize later.

However i think you should add the db transaction back in, otherwise this single function will run over 100 queries.

Reply:
I'll do that shortly.